src/main/kotlin/com/sipios/springsearch/strategies/ParsingStrategy.kt

```diff
@@ -61,7 +62,15 @@ interface ParsingStrategy {
                 Float::class -> FloatStrategy()
                 Int::class -> IntStrategy()
                 Date::class -> DateStrategy()
-                else -> StringStrategy(searchSpecAnnotation)
+                else -> getStrategyWithSuperClass(fieldClass, searchSpecAnnotation)
+            }
+        }
+
+        private fun getStrategyWithSuperClass(fieldClass: KClass<out Any>, searchSpecAnnotation: SearchSpec): ParsingStrategy {
+            return if (fieldClass.superclasses[0] == Enum::class) {
```

Comment:
instead of using superClasses can we use [isSubclassOf](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.reflect.full/is-subclass-of.html) ?

Reply:
Good suggestion, I changed :)